Uncovered apm(4) ioctl(2) prevents pledge(2) usage.
unveil(2) all files read-write-create to prevent execution.

Index: src/up-main.c
--- src/up-main.c.orig
+++ src/up-main.c
@@ -22,6 +22,7 @@
 
 #include "config.h"
 
+#include <err.h>
 #include <string.h>
 #include <signal.h>
 #include <sys/types.h>
@@ -184,6 +185,14 @@ main (gint argc, gchar **argv)
 	GBusNameOwnerFlags bus_flags;
 	gboolean replace = FALSE;
 
+	if (unveil("/", "r") == -1)
+		err(1, "unveil /");
+	if (unveil("/var/run/dbus", "rw") == -1)
+		err(1, "unveil /var/run/dbus");
+	if (unveil("/var/db/upower", "rwc") == -1)
+		err(1, "unveil /var/db/upower");
+	if (unveil(NULL, NULL) == -1)
+		err(1, "unveil NULL");
 	const GOptionEntry options[] = {
 		{ "timed-exit", '\0', 0, G_OPTION_ARG_NONE, &timed_exit,
 		  /* TRANSLATORS: exit after we've started up, used for user profiling */
